device/mellanox/x86_64-mlnx_msn2700-r0/psu_sensors_conf_updater

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+function update_psu_sensors_configuration() {
+    local SENSORS_CONFIG_FILE="/tmp/sensors.conf"
+    local PSU_SENSORS_CONFIG="/usr/share/sonic/platform/psu_sensors.conf"
+
+    if [ -f $SENSORS_CONFIG_FILE ]; then
+        rm $SENSORS_CONFIG_FILE
+    fi
+
+    if [ -z "$1" ]; then
+        echo "ERROR: no sensors configuration file had been selected"
+        return
+    else
+        cp $1 $SENSORS_CONFIG_FILE
+    fi
+
+    platform=$(cat /usr/share/sonic/platform/platform.json | grep -m 1 "name" | cut -d '"' -f 4)
+    # built-in psu, no need for dynamic configruation
+    if [[ "$platform" == "MSN2100" || "$platform" == "MSN2010" ]]; then return; fi
+    # incase we don't have this platform's info
+    if [[ $(jq -r --arg platform "$platform" '.platform[$platform]' PSU_SENSORS_CONFIG) == "null" ]]; then return; fi
+
+    local rev=""
+    case $(echo $1 | rev | cut -d/ -f 1| rev) in 
```

Comment:
maybe we need to find a more graceful way to handle the files, it someone changes the file name or adds a new file, then the scripts will be broken.

Reply:
add related hardware revision marker in the sensors.conf, and change this logic to directly extract the rev from it